akka-javasdk/src/main/java/akka/javasdk/agent/SessionMemoryEntity.java

```diff
@@ -82,10 +83,16 @@ public State clear() {
     private static long enforceMaxCapacity(List<SessionMessage> messages, long currentLengthSize, long maxSize) {
       var freedSpace = 0;
       while ((currentLengthSize - freedSpace) > maxSize) {
-        // FIXME: delete also the reply from AI?
         freedSpace += messages.removeFirst().size();
         logger.debug("Removed oldest message. Remaining size={}, maxSizeInBytes={}", currentLengthSize, maxSize);
       }
+
+      // remove all messages that are not UserMessage since those were driven by the deleted UserMessage
+      while (!messages.isEmpty() && !(messages.getFirst() instanceof UserMessage)) {
+        freedSpace += messages.removeFirst().size();
+        logger.debug("Removed orphan message. Remaining size={}, maxSizeInBytes={}", currentLengthSize, maxSize);
```

Comment:
```suggestion
        logger.debug("Removed orphan message. Remaining size [{}], maxSizeInBytes [{}]", currentLengthSize, maxSize);
```

Reply:
some other logging here that is not following that style

Would probably be useful to include the session id in all logging

Reply:
Changed now to log sessionId and use the format: `... sessionId [{}]. Remaining size [{}], maxSizeInBytes [{}]",`

I have been wondering though, is there a reason to use this format instead of sessionId=[{}] ? The reason I ask this is: from my experience the log processing platforms usually automatically extract keyvalues ([example](https://dev.splunk.com/enterprise/docs/developapps/addsupport/logging/loggingbestpractices/#Use-clear-key-value-pairs)), so one could automatically search for `sessionId=X` on the logs.. but with this format you can only search by `X`. It does not matter much for the 2 last values but for things like IDs can be useful I think. Am I missing something?

Reply:
Might not be the most optimal pattern, but better to stick to the same convention for all Akka logging. It has been used for 10+ years and wouldn't be easy to change everywhere.